src/common/tusb_compiler.h

```diff
@@ -138,10 +138,14 @@
   #define TU_ATTR_BIT_FIELD_ORDER_BEGIN
   #define TU_ATTR_BIT_FIELD_ORDER_END
 
-  #if __has_attribute(__fallthrough__)
-    #define TU_ATTR_FALLTHROUGH         __attribute__((fallthrough))
-  #else
+  #if defined(__XC16)
```

Comment:
__has_attribute() is introduced since gcc version 5. we could replace XC16 with more generic `__GNUC__ < 5`